reference/shaders-msl/comp/global-invocation-id-writable-ssbo-in-function.comp

```diff
@@ -0,0 +1,31 @@
+#pragma clang diagnostic ignored "-Wmissing-prototypes"
+
+#include <metal_stdlib>
+#include <simd/simd.h>
+
+using namespace metal;
+
+struct myBlock
+{
+    int a;
+    float b[1];
+};
+
+// Support GLSL mod(), which is slightly different than Metal fmod()
+template<typename Tx, typename Ty>
+Tx mod(Tx x, Ty y)
+{
+    return x - y * floor(x / y);
+}
+
+float getB(device const myBlock& myStorage, thread const uint3& gl_GlobalInvocationID)
```

Comment:
What is causing this bug? It doesn't seem too hard to not add "const" if "device" storage is used?